src/ishikawa_backend.erl

```diff
@@ -125,16 +142,67 @@ handle_call({tcbcast, Message}, _From, #state{actor=Actor,
     %% Transmit to membership.
     [send(Message, Peer) || Peer <- Members],
 
-    {reply, ok, State#state{vv=VClock}};
-handle_call({tcbdeliver, _Message, _Timestamp}, _From, State) ->
-    %% TODO: Implement me.
-    {reply, ok, State};
+    %% Add members to the queue of not ack messages.
+    ToBeAckQueue1 = [ToBeAckQueue | [{Message, Members}]],
+
+    {reply, ok, State#state{vv=VClock, to_be_ack_queue=ToBeAckQueue1}};
+handle_call({tcbdeliver, Origin, Message, Timestamp}, _From, #state{vv=VClock0,
+                                              rtm=RTM0,
+                                              to_be_delivered_queue=Queue0} = State) ->
+
+    %% Check if the message should be delivered
```

Comment:
I'm not sure what this is doing.

What's the expected behavior of `check_causal_delivery`?  The way this comment is worded, it sounds like it's supposed to be a boolean?